### PR TITLE
Perform a basic HTTP request for health check

### PR DESCRIPTION
**Pull request by nolancon**
### Description of your changes
Replace the existing health check mechanism (create a designated health check bucket and perform put/get object calls) with a basic HTTP request to the host base. The existing mechanism is complex and fragile - we have seen issues related to health check bucket ownership. A simple HTTP ping will suffice instead.

Fixes #

I have:

- [ ] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested
CI + kuttl/ceph-kuttl tests (changes made to both)

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: internal/controller/providerconfig/healthcheck/healthcheck_controller.go [hunk 1]
```diff
@@ -199,70 +158,60 @@ func (c *Controller) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
...
 	}
 
-	_, getErr := rgw.GetObject(ctx, s3BackendClient, &s3.GetObjectInput{
-		Bucket: aws.String(bucketName),
-		Key:    aws.String(healthCheckFile),
-	}, o...)
-	if getErr != nil {
-		getErr = errors.Wrap(getErr, errGetHealthCheckFile)
-		traces.SetAndRecordError(span, getErr)
+	resp, err := c.httpClient.Do(req)
```
Comment (mhmxs) [inline internal/controller/providerconfig/healthcheck/healthcheck_controller.go:195]: As i see it would use the default timeout of go: `0`

**Comment by mhmxs** [comment]
@nolancon Does it make sense to replace "is bucket exists" calls to http in the bucket controller?

File: internal/controller/providerconfig/healthcheck/healthcheck_controller.go (hunk 1 above)
Comment (nolancon) [reply internal/controller/providerconfig/healthcheck/healthcheck_controller.go:195]: :+1:  changed it to use S3 timeout - which is what the original health check would have been using.

**Comment by nolancon** [comment]
> @nolancon Does it make sense to replace "is bucket exists" calls to http in the bucket controller?

I'm not sure. I like the fact that we get a specific error from it as opposed to a simple http request, but I don't feel very strongly on it. If so, I think it would deserve its own PR.

The main reason for changing the health check was because we have this single long-living health-check bucket that we need to write to and it has caused us issues with permissions - whereas that's not really an issue for generic buckets.

**Review by mhmxs — APPROVED** [review]

**Pull request merged**